pkg/pquerier/querier.go

```diff
@@ -321,12 +321,16 @@ func (q *V3ioQuerier) GetLabelSets(metric string, filter string) ([]utils.Labels
 
 	// Get all label sets
 	input := v3io.GetItemsInput{
-		Path:           partitionPaths[0],
 		Filter:         filter,
 		AttributeNames: []string{config.LabelSetAttrName, config.MetricNameAttrName},
 	}
 
-	iter, err := utils.NewAsyncItemsCursor(q.container, &input, q.cfg.QryWorkers, shardingKeys, q.logger)
+	// because of performance issues we only want to query the first and last partitions
+	partitionsToQuery := []string{partitionPaths[len(partitionPaths)-1]}
```

Comment:
can partitions be empty? then it's size is 0

Reply:
https://github.com/v3io/v3io-tsdb/pull/476/files#diff-44f0b6ed4756e6b5b32c5a5b19a6d5d4R311